kylinpy/service/kylin_service.py

```diff
@@ -112,6 +112,70 @@ def query(self, sql, limit=50000, offset=0, acceptPartial=False, **kwargs):
 
         return response
 
+    def query_prestate(self, sql, limit=0, offset=50000, acceptPartial=False, **kwargs):
+        """https://cwiki.apache.org/confluence/display/KYLIN/Query+API"""
+        json_data = {
+            'acceptPartial': acceptPartial,
+            'limit': limit,
+            'offset': offset,
+            'project': self.project,
+            'sql': sql,
+        }
+        kwargs.setdefault('json', json_data)
+        try:
+            response = self.client.post('/query/prestate', **kwargs).json()
+        except InternalServerError as err:
+            raise KylinQueryError(err)
+
+        return response
+
+    def create_saved_queries(self, sql, name, desc=None, **kwargs):
+        json_data = {
+            'sql': sql,
+            'name': name,
+            'project': self.project,
+            'description': desc,
+        }
+        kwargs.setdefault('json', json_data)
+        try:
+            response = self.client.post('/saved_queries', **kwargs).json()
+        except InternalServerError as err:
+            raise KylinQueryError(err)
+
+        return response
```

Comment:
Hi @yqkcn, thanks for adding new API support in Kylinpy, I'll add the first quick pass review.

The `KylinQueryError` refers to a query that is incorrect or unreachable so a normal API shouldn't be raised for that one. 

On the other hand, the `KylinQueryError` also be used in catching a bad query on the upstream use case.

Reply:
OK, I will add some Error Inherit from `KylinQueryError`, and also add some unit tests.